Tools/ardupilotwaf/ardupilotwaf.py

```diff
@@ -204,6 +204,26 @@ def ap_find_tests(bld, use=[]):
             cxxflags=['-Wno-undef'],
         )
 
+_versions = []
+
+@conf
+def ap_version_append(ctx, k, v):
+    ctx.env['AP_VERSION_ITEMS'] += [(k, os.environ.get(k, v))]
+
+@conf
+def ap_version_append_str(ctx, k, v):
+    ctx.env['AP_VERSION_ITEMS'] += [(k, '"{}"'.format(os.environ.get(k, v)))]
+
+@conf
+def write_version_header(ctx):
```

Comment:
Hm, I think this should take an optional argument for the filename. And [here](https://github.com/ArduPilot/ardupilot/pull/4053/commits/7823173785bcdb286a2c022222b6eb88d54e613f#diff-dc6134e3d3c869a052520d86f46371dcR256) we should use a wrapper calling this function with `self.target`. See "Rule functions" in https://waf.io/book/#_rule_based_task_generators_make_like.